tinytuya/__init__.py

```diff
@@ -263,28 +263,55 @@ def __init__(self, dev_id, address, local_key="", dev_type="default", connection
         self.retry = True
         self.dev_type = dev_type
         self.port = 6668  # default - do not expect caller to pass in
+        self.socket=None
+
+    def __del__(self):
+        if self.socket!=None:
+            self.socket.close()
+            self.socket=None
 
     def __repr__(self):
         return '%r' % ((self.id, self.address),)  # FIXME can do better than this
 
+    def _get_socket(self,renew):
+        if(renew and self.socket!=None):
+            self.socket.close()
+            self.socket=None
+        if(self.socket==None):
+          self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+          #s.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)

```

Comment:
I'm not sure if some devices need NODELAY.  
I think I got fewer 10054 after commenting it out.  
Let me know if you want me to make a branch without this change.